samza-core/src/main/java/org/apache/samza/config/ConfigLogger.java

```diff
@@ -0,0 +1,71 @@
+package org.apache.samza.config;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.util.function.Consumer;
+
+/**
+ * This logger logs the config line by line, while shortening the lines which are too long (over maxLen)
+ * Logging preserves the original logger name and may be redirected to a different file
+ * by specifying a logger name that starts with ConfigLogger. in log4j configuration.
+ */
+public class ConfigLogger {
```

Comment:
What's the purpose of this class? Where is this used?

I'd strongly prefer to not do this. IMHO it'll be much better to log these things (config, job model, starting offsets) as JSON instead, so that the log representation is compact but we can copy and "pretty print" in the IDE / browser for explorability.